controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -107,3 +120,161 @@ func (r *Reconciler) addNewTierUpdate(ctx context.Context, tier *toolchainv1alph
 	})
 	return r.Client.Status().Update(ctx, tier)
 }
+
+// ensureRevision ensures that there is a TierTemplateRevision CR for each of the TierTemplate.
+// returns `true` if a new TierTemplateRevision CR was created, `err` if something wrong happened
+func (r *Reconciler) ensureRevision(ctx context.Context, nsTmplTier *toolchainv1alpha1.NSTemplateTier) (bool, error) {
+	logger := log.FromContext(ctx)
+	refs := getNSTemplateTierRefs(nsTmplTier)
+
+	// init revisions
+	if nsTmplTier.Status.Revisions == nil {
+		nsTmplTier.Status.Revisions = map[string]string{}
+	}
+	// check for TierTemplates and TierTemplateRevisions associated with the NSTemplateTier
+	ttrCreated := false
+	for _, tierTemplateRef := range refs {
+		// get the TierTemplate
+		var tierTemplate toolchainv1alpha1.TierTemplate
+		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: nsTmplTier.GetNamespace(), Name: tierTemplateRef}, &tierTemplate); err != nil {
+			// something went wrong or we haven't found the TierTemplate
+			return false, err
+		}
+
+		// check if there is TTR associated with this TierTemplate
+		ttrCreatedLatest, err := r.ensureTTRforTemplate(ctx, nsTmplTier, &tierTemplate)
+		ttrCreated = ttrCreated || ttrCreatedLatest
```

Comment:
if there is a case when the value of  `ttrCreated`  is false  and `ttrCreatedLatest` is true, wanted to understand how that would be still okay here?

Reply:
Basically this logic ensures that `ttrCreated` is set to true at the end if at least one TTR was created. Thus if `false || true` will return `true` from your example. Hope this makes sense.